### PR TITLE
PLANET-7573: Apply necessary changes to update ElasticPress to version 5.*

### PR DESCRIPTION
**Pull request by mardelnet**
Ref: https://jira.greenpeace.org/browse/PLANET-7573 

<hr>

**DESCRIPTION:**

This PR:
- Updates the links to the ElasticPress sync page, according to the changes implemented in version 5 of this plugin.
- Conditionally adds/hides the `admin.php?page=planet4_settings_search_content` page if ElasticPress is active.
- Updates the search tests.
 
<hr>

**RELATED PR:**

https://github.com/greenpeace/planet4-base/pull/307

<hr>

**POSSIBLE CHANGES:**

Should we conditionally add/hide the `admin.php?page=planet4_settings_search_content` page if ElasticPress is active, @comzeradd ?

### CODE REVIEW

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [nix](https://www-dev.greenpeace.org/test-nix) | [admin](https://www-dev.greenpeace.org/test-nix/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-nix/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-nix) | [composer-local.json](https://github.com/greenpeace/planet4-test-nix/blob/main/composer-local.json)

:watch: 2024.08.14 08:22:22

**Comment by comzeradd** [comment]
> Should we conditionally add/hide the `admin.php?page=planet4_settings_search_content` page if ElasticPress is active, @comzeradd ?

If you think that's an easy addition (since ElasticPress is practically activated on all production sites) then yes, that would be a nice enhancement.

**Comment by mardelnet** [comment]
> If you think that's an easy addition (since ElasticPress is practically activated on all production sites) then yes, that would be a nice enhancement.

Done,@comzeradd

File: src/Settings.php [hunk 1]
```diff
@@ -503,6 +459,57 @@ public function __construct()
...
+            $search_content = array(
+                'planet4_settings_search_content' => [
+                    'title' => 'Search content',
+                    'fields' => [
+                        [
+                            'name' => __('Include archived content in search for', 'planet4-master-theme-backend'),
+                            'desc' => __(
+                                '<b>Important:</b> On change of Include archive content setting, Please kindly',
+                                'planet4-master-theme-backend'
+                            ) . ' <a href="admin.php?page=elasticpress-sync" target="_blank">Sync Elasticsearch</a>.',
```
Comment (comzeradd) [inline src/Settings.php:474]: nit: we can remove the `target="_blank"` here

**Review by comzeradd — APPROVED** [review]
Nice 👍🏻

File: src/Settings.php (hunk 1 above)
Comment (mardelnet) [reply src/Settings.php]: Good idea, done!

**Comment by mardelnet** [comment]
@comzeradd:
The deployment of the test instance [was failing to build](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-nix/1120/workflows/cb88c82d-1c95-429b-9698-1289c6f21913/jobs/3963/parallel-runs/0/steps/0-111)  because of [renaming `facets` as `filters`](https://github.com/greenpeace/planet4-base/pull/307#issuecomment-2288496618) as you can see [here](https://github.com/greenpeace/planet4-master-theme/pull/2347/commits/3c3556397efe1feb9f128aaf772110559dbbc4db), so I had to revert that change.
I also updated one of the E2E tests. Since upgrading ElasticPress requires a full fresh sync, the tag names were not shown on the results page (more info [here](https://greenpeace.slack.com/archives/C014UMRC4AJ/p1723552295828479)) and this line of the tests was failing: `expect(searchTags).toContain(`#${tagName}`);`

**Comment by mardelnet** [comment]
Closing this PR in favour of https://github.com/greenpeace/planet4-master-theme/pull/2354

**Pull request closed without merging**